pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -0,0 +1,618 @@
+                                                handleSegmentFuture.completeExceptionally(exception);
+                                            }
+                                        }, null);
+                            });
+                            openManagedLedgerFuture.complete(null);
+                        }
+
+                        @Override
+                        public void openReadOnlyManagedLedgerFailed(ManagedLedgerException exception, Object ctx) {
+                            log.error("[{}] Failed to open readOnly managed ledger", topic, exception);
+                            openManagedLedgerFuture.completeExceptionally(exception);
+                        }
+                    };
+
+                    TopicName snapshotIndexTopicName = TopicName.get(TopicDomain.persistent.toString(),
+                            TopicName.get(topic.getName()).getNamespaceObject(),
+                            EventType.TRANSACTION_BUFFER_SNAPSHOT_SEGMENTS.toString());
+                    this.topic.getBrokerService().getPulsar().getManagedLedgerFactory()
+                            .asyncOpenReadOnlyManagedLedger(snapshotIndexTopicName
+                                            .getPersistenceNamingEncoding(), callback,
+                                    topic.getManagedLedger().getConfig(),
+                                    null);
+                    //Wait the processor recover completely and the allow TB to recover the messages
+                    // after the startReadCursorPosition.
+
+                    return openManagedLedgerFuture
+                            .thenCompose((ignore) -> FutureUtil.waitForAll(completableFutures).thenCompose((i) -> {
+                                if (hasInvalidIndex.get()) {
+                                    persistentWorker.appendTask(PersistentWorker.OperationType.UpdateIndex, ()
+                                            -> persistentWorker
+                                            .updateSnapshotIndex(persistentSnapshotIndexes.getSnapshot(),
+                                            indexes.values().stream().toList()));
+                                }
+                                return CompletableFuture.completedFuture(finalStartReadCursorPosition);
+                            })).exceptionally(ex -> {
+                                log.error("[{}] Failed to recover snapshot segment", this.topic.getName(), ex);
+                                return null;
+                            });
+
+                    },  topic.getBrokerService().getPulsar().getTransactionExecutorProvider()
+                        .getExecutor(this));
+    }
+
+    @Override
+    public CompletableFuture<Void> deleteAbortedTxnSnapshot() {
+        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+        persistentWorker.appendTask(PersistentWorker.OperationType.Close,
```

Comment:
Because this operation is executed in `transactionBufferCleanupAndClose`so I name it as close at the beginning.
But the `clear` may be better for the operation type of PersistentWorker.